spec/rubocop/cop/rspec/describe_class_spec.rb

```diff
@@ -215,4 +215,23 @@
       RUBY
     end
   end
+
+  context 'with mixin tests' do
+    describe '#top_level_group?' do
+      subject(:cop_with_top_level_group_mixin) do
+        described_class.new
+      end
+
+      it 'warns because it is deprecated' do
+        # rubocop:disable RSpec/SubjectStub
+        allow(cop_with_top_level_group_mixin).to receive(:warn)
+        allow(cop_with_top_level_group_mixin).to receive(:top_level_groups)
+          .and_return([])
+
+        cop_with_top_level_group_mixin.send(:top_level_group?, nil)
+        expect(cop_with_top_level_group_mixin).to have_received(:warn)
+        # rubocop:enable RSpec/SubjectStub
+      end
+    end
+  end
```

Comment:
We can add a new file testing only the module you changed – and that way we can (almost) avoid stubbing. E.g.:
```rb
# spec/rubocop/cop/rspec/mixin/top_level_group_spec.rb

RSpec.describe RuboCop::Cop::RSpec::TopLevelGroup do
  describe '#top_level_group?' do
    let(:stub_class) do
      Class.new do
        include RuboCop::Cop::RSpec::TopLevelGroup

        def initialize
          @top_level_groups = []
        end

        def test_top_level_group
          top_level_group?(nil)
        end
      end
    end

    it 'warns because it is deprecated' do
      expect { stub_class.new.test_top_level_group }.to \
        output(/warning: top_level_group\? is deprecated/).to_stderr
    end
  end
end
```